### PR TITLE
Linstor fix migration while node offline

### PR DESCRIPTION
**Pull request by rp-**
### Description

If a Linstor node is down while migrating resource, allow-two-primaries
setting will fail because we can't reach the downed node. But it will
still set the property on the other nodes and migration should work.
We now just report an error instead of completely failing.

This PR also containes a cherry-picked commit from the main branch f176e7d122be0a9fa448dea547aa2d2e0e867352

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

Nested Linstor cluster with a non hyperconverged setup and a node evicted.

### CODE REVIEW

**Comment by rp-** [comment]
Commit: 1f6954c7cc5955c21de3c09131991554e5772fc5 should also be applied to 4.19 and main
But I can make a separate PR for that?

File: plugins/storage/volume/linstor/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStorageAdaptor.java [hunk 1]
```diff
@@ -353,19 +361,21 @@ public boolean disconnectPhysicalDiskByPath(String localPath)
...
                 }
                 s_logger.warn("Linstor: Couldn't find resource for this path: " + localPath);
             } catch (ApiException apiEx) {
-                s_logger.error(apiEx);
-                throw new CloudRuntimeException(apiEx.getBestMessage(), apiEx);
+                s_logger.error(apiEx.getBestMessage());
+                // do not fail here as removing allow-two-primaries property isn't fatal
             }
         }
-        return false;
```
Comment (sureshanaparti) [inline plugins/storage/volume/linstor/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStorageAdaptor.java:368]: @rp- any case returns false (on error / exception above ?)

**Review by sureshanaparti — APPROVED** [review]
changes lgtm

**Comment by sureshanaparti** [comment]
> Commit: [1f6954c](https://github.com/apache/cloudstack/commit/1f6954c7cc5955c21de3c09131991554e5772fc5) should also be applied to 4.19 and main But I can make a separate PR for that?

Entire PR changes can be forward merged to 4.19 / main. I think, merge excludes other changes if there are already in.

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by rohityadavcloud** [comment]
Merged based on smoketests. All changes in Linstor plugin.

**Pull request merged**